File: C/CWE-477/1/bad.c
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */
#include <stdio.h>
#include <stdlib.h>
#include <stdatomic.h>

int main(int argc, char *argv[])
{
	atomic_int i = ATOMIC_VAR_INIT(1); //ATOMIC_VAR_INIT deprecated by C17 standard
	printf("%d.\n", i);
	return EXIT_SUCCESS;
}

File: C#/CWE-477/1/bad.cs
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */

using System;

namespace BadWeakness
{
	class Bad
	{
		static void Main()
		{
			Deprecated.Deprecate("Testing Function…"); //call to deprecated function
		}
	}

	static class Deprecated
	{
		[Obsolete("Switch to using DeprecateLine(string s) instead.")]
		public static void Deprecate(string s)
		{
			Console.Write(s + Environment.NewLine);
		}

		public static void DeprecateLine(string s)
		{
			Console.WriteLine(s);
		}
	}
}
